Technical-Steering-Committee/best-practices/contrinuous-integration.md

```diff
@@ -0,0 +1,23 @@
+# Continuous Integration
+
+## Prerequisites
+All Zowe repositories containing source code that requires compilation, packaging, testing or other type of verification, should define a continuos integration process that uses FLOSS automation tools, free to acquire and use for any purposes. It is recommended to choose tools already popular within Zowe, such as Gradle, npm, GitHub Actions, etc. For specific needs, it is possible to select alternatives, given they provide unique and necessary advantages.
+
+## Compilation
+If a Zowe repository requires source code compilation, this repository should include compilers, scripts and tools that automate the process, handle dependencies and produce project artifacts.
+
+## Packaging
+If a Zowe module is intended to be delivered as a deployable installation packed, its corresponding repositories should contain scripts and tools that group the necessary  project artifacts included or produced by compilation into a single or multiple packages targeted for their deployment environments.
+
+## Testing
+Every Zowe module or repository should contain a holistic and automated test suit that is executed on every code change or repository build, i.e. compilation and packaging. Ideally the automated test suit should consist of unit tests to cover the modules being developed or modified, as well as integration tests and system tests to cover interactions with dependent systems. The tests should cover both functional, i.e. validating its functionality, as well as non-functional, i.e. validating performance, reliability, resilience, etc. 
+
+## Code Analysis
+It is required, whenever reasonable, each repository to make use of static code analysis and vulnerability scans per tools availability and requirement. The recommended tools are SonarCloud and CodeQL and should be configured to perform scans for each change. All reported issues should be addressed either by resolving them or by indicating an exception justification.
```

Comment:
1)IMO it is always reasonable to perform code analysis. What exceptions are justifiable to not do so?
2) A minor suggestion: It would more precise to say that the pipelines should be configured to run the corresponding scanning tools instead of the tools themselves to be intercepting the buildin pipelines.

```suggestion
It is required, each repository to make use of static code analysis and vulnerability scans per tools availability and requirement. The recommended tools are SonarCloud and CodeQL and the build pipelines should be configured to run the tools and perform scans for each change. All reported issues should be addressed either by resolving them or by indicating an exception justification.
```

Reply:
Resolved in the latest commit.